vendor/plugins/edu.harvard.dce.paella.timedCommentsPlugin/timedCommentsFootprints.js

```diff
@@ -0,0 +1,292 @@
+// Rute's comment
+Class ("paella.plugins.timedCommentsHeatmapPlugin", paella.ButtonPlugin, {
+  INTERVAL_LENGTH: 5,
+  inPosition: 0,
+  outPosition: 0,
+  canvas: null,
+  commentHeatmapTimer: null,
+  footPrintData: {},
+  commentTotal: 0,
+  ifModifiedSinceDate: "1999-12-31T23:59:59Z", //yyyy-MM-dd'T'HH:mm:ss'Z',
+  isEnabled: false,
+
+  getAlignment: function () {
+    return 'right';
+  },
+  getSubclass: function () {
+    return "commentHeatmap comments";
+  },
+  getIndex: function () {
+    return 590;
+  },
+  getDefaultToolTip: function () {
+    return base.dictionary.translate("Show comments");
+  },
+  getName: function () {
+    return "es.upv.paella.timedCommentsHeatmapPlugin";
+  },
+  getButtonType: function () {
+    return paella.ButtonPlugin.type.timeLineButton;
+  },
+
+
+  setup: function () {
+    var thisClass = this;
+
+    switch (this.config.skin) {
+      case 'custom':
+      this.fillStyle = this.config.fillStyle;
+      this.strokeStyle = this.config.strokeStyle;
+      break;
+
+      case 'dark':
+      this.fillStyle = '#727272';
+      this.strokeStyle = '#424242';
+      break;
+
+      case 'light':
+      this.fillStyle = '#d8d8d8';
+      this.strokeStyle = '#ffffff';
+      break;
+
+      default:
+      this.fillStyle = '#d8d8d8';
+      this.strokeStyle = '#ffffff';
+      break;
+    }
+  },
+
+  checkEnabled: function (onSuccess) {
+    var thisClass = this;
+    paella.data.read('timedComments', {
+      id: paella.initDelegate.getId(),
+      question: 'canAnnotate'
+    }, function(data) {
+      base.log.debug("TC canAnnotate " + data);
+      if (data === "true") {
+        // prevent annots on live stream for now, until test live video inpoints
+        thisClass.isEnabled = !paella.player.isLiveStream();
+        onSuccess(!paella.player.isLiveStream());
+      } else {
+        onSuccess(false);
+      }
+    });
+  },
+
+  buildContent: function (domElement) {
+    var container = document.createElement('div');
+    container.className = 'commentHeatmapContainer';
+
+    this.canvas = document.createElement('canvas');
+    this.canvas.id = 'commentHeatmapCanvas';
+    this.canvas.className = 'commentHeatmapCanvas';
+    container.appendChild(this.canvas);
+
+    
+    domElement.appendChild(container);
+  },
+
+  willShowContent: function () {
+    var thisClass = this;
+    thisClass.loadcommentHeatmap();
+    base.log.debug("TC showing comments heatmap" + new Date());
+    // new event type created by timedCommentsOverlayPlugin
+    if (paella.events.showTimedComments) {
+      paella.events.trigger(paella.events.showTimedComments, {
+        sender: this
+      });
+    }
+    thisClass.commentHeatmapTimer = new base.Timer(function (timer) {
+      thisClass.loadcommentHeatmap(true);
+    },
+    5000);
+    thisClass.commentHeatmapTimer.repeat = true;
+  },
+
+  // on refreshes data if result TOTAL has changed since last update
```

Comment:
Should total still be used? What if there's a delete (by admin) and an insert at different points?